### DISCUSSION TITLE
Latest and Greatest: Control v0.0.7 and OS v0.0.7 Updates

### DISCUSSION THREAD

**Discussion by ntorga** [goinfinite/os#172] [Announcements]
## [Speedia Control](https://speedia.net/control/) v0.0.7

This month brings a game-changing update to [Speedia Control](https://speedia.net/control/), introducing powerful features that significantly enhance your container management experience.

### • Container Images

We're thrilled to introduce container snapshots and image management capabilities:

![container-images-screenshot](https://github.com/user-attachments/assets/45730ca3-9382-4975-bf59-ceeb10b7d5a6)

We've added the ability to take container snapshots, import container images, and manage them on the server. You can even archive images with your preferred compression tool for download to your local machine:

![container-image-take-snapshot](https://github.com/user-attachments/assets/aaaf33de-235e-442a-b776-96cc3176e6f1)

This feature bridges the gap between local development and cloud deployment. Work on projects locally, then effortlessly transfer them to Speedia Control for collaboration or production use. It's a major time-saver that we're particularly proud of, given its complex implementation:

![container-image-import-container](https://github.com/user-attachments/assets/2c5f06f5-0f46-46e9-a193-8342a2d0173c)

### • Security Records for All Write Operations

We've also implemented the security record feature to all write operations. This means you can keep track of who's doing what, when, and to which resources. Transparency is key, after all. With this feature, you'll have a clear audit trail of all changes made to your platform, giving you peace of mind and helping you stay on top of security.

Of course, no update would be complete without a few bug squashes and minor tweaks. We've:

- fix: duplicated SSL map block bug that was causing issues for some users;
- fix: sorted out a network parsing issue with newer Podman versions;
- fix: remove the textual form on modals on container profiles page;
- fix: added blank metrics for containers that aren't running to prevent display issues.

## [Speedia OS](https://speedia.net/os/) v0.0.7

Speedia OS has had a bit of a makeover this month, with some exciting new features and improvements.

### • OpenSearch

You can now finally install OpenSearch using the "Install Service" modal or the CLI which  opens up a whole new world of possibilities. OpenSearch is a powerful search and analytics engine that can help you make sense of your container data.

### • OpenMage & Adobe Commerce

With OpenSearch on board, we've added OpenMage and Adobe Commerce to the marketplace. This is just the beginning, so stay tuned for more additions in the future. Our marketplace is designed to make it easy to find and install the tools and services you need to get the most out of your containers.

![install-openmage](https://github.com/user-attachments/assets/6a272863-bcc3-44ed-98b4-b8208d4aa0c0)

### • Java Support

We've also added support for Java applications, with a range of vendors to choose from, including OpenJDK, Temurin, Corretto, AdoptOpenJdk, Zulu, IBM Semeru, SapMachine, Oracle, GraalVM, Microsoft, and Liberica. This should make life a lot easier for all you Java developers out there. With Speedia OS, you'll be able to easily manage your Java applications, from deployment to scaling and monitoring.

### • Refactored Interfaces: Mappings, Database and SSL Certificates

As part of our ongoing effort to refactor the front-end, we've rewritten the mapping, database, and SSL interfaces using HTMX and Alpine.js. It's all part of our mission to make Speedia OS the best it can be and fully open-source, including the front-end.

![mappings-mgmt](https://github.com/user-attachments/assets/1f948d74-9104-4f35-889c-eed4368785df)
![database-mgmt](https://github.com/user-attachments/assets/a395ba5d-98c0-46b8-a10a-b762f76dc60a)
![ssl-mgmt](https://github.com/user-attachments/assets/9a614b4a-e01d-4a82-b413-6468f9b43ffa)

And, as always, we've squashed a few bugs and made some minor improvements, including:

- fix: fix response code bug when creating mapping;
- fix: reduce proxy buffers to reasonable levels;
- fix: add admin pass length check to marketplace items.

**Comment by queirvz** [comment]
---
Way to go, Northon. Keep up the good work! 🚀
---

First, I’d like to say this release introduces some highly useful features, particularly for teams working on rapid container deployments. The new image management tools and security audit trail are valuable additions that can certainly streamline operations for businesses scaling their containerized environments -- specially in SMBs in need of cost-effective yet secure resource-management solutions.

I enjoyed exploring the platform and have some suggestions to enhance deployment availability, as well as additional ideas for further optimization.

## Container Image Management and the Audit Trail

The introduction of snapshots and image compression is a smart move. We can make it easier to manage large datasets. To further enhance this, I'd recommend adding a feature to automatically schedule snapshots and compressions based on user-defined policies. This could  dimerize the platform's value proposition as resources are scaled. You see all assets are subject to feedback mechanisms in the form of logs and audit trails, so why not leverage this data to provide users with insights on their storage usage and potential savings? You started with images, and think about other heavy assets like DBs and logs. Maybe we'll need some greping and piping to self-hosted natural language processing services to make sense of it all.

## Security Records for Write Operations

The audit trail feature is fantastic for transparency. Imagine providing granular user access controls as part of the Control tiered API -- this could be a valuable add-on for businesses with strict compliance requirements or those managing sensitive data. The Java ecosystem support consuming the Control API agnostically from Go and other languages could be a great way to show we truly care about making something teams will use.

## Marketplace Expansion

Integrating OpenSearch and commerce platforms is a great step toward adding value for users. Offering one-click deployment of popular e-commerce stacks and coupling them with managed support packages is a way to handle the convenience and peace of mind businesses seek when deploying their production environments. The community images in JSON, YAML, TOML or other standard machine-readable formats are a great start, and expanding this to include more specialized stacks may enhance Infinite Cloud's use-case flexibility. With this, you could also consider adding a feature to allow users to submit their own images to the marketplace. Should this be pushed to a public registry, it could remove the hassle of managing custom images.


## Refactored Interfaces (HTMX & Alpine.js)

The UI refactor using HTMX and Alpine.js is a step in making the platform more responsive and intuitive. This is a great foundation for further enhancements, such as adding real-time notifications for container status changes or resource usage alerts. This could be a free forever feature, as it would provide users with a more interactive experience and help them stay on top of their deployments. The goodie bag of features is a great way to drive adoption. File management and the ability to upload and download files directly from the browser is a great feature that could be further enhanced with the drag-and-drop functionality you had in mind. Wow. Can't wait for the shortcuts kit as well: `J` + `K` for navigation, `/` for searching through deployed images, and snappy actions for container management. A dream come true.

As for additional Ideas for Optimization:

- Usage Analytics: Adding a feature to provide users with container utilization and performance metrics, perhaps integrated with OpenSearch, could lead to better insights for businesses.
- SaaS Integration: Ease of use could be driven even further by offering integrations with billing systems or external SaaS apps. Nice call on the Stripe integration, by the way. This could allow users to automatically monitor and charge for their container usage, especially relevant for users running multi-tenant applications. Pro-active monitoring and billing is a way to make users feel more in control of their deployments -- and their pockets. So next time they deploy a container, they know exactly how much it will cost them based on average usage and projected growth. Forecasting based on variance and volatility of the images resource consumption is another step in the right direction -- in the future, I guess.
	
Congratulations on the release!
	
GQ
